spec/migrations/remove_general_prefix_setting_spec.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+mig_file = Dir[Rails.root.join('db/migrate/20210628134317_remove_general_prefix_settings.rb')].first
+require mig_file
+
+describe RemoveGeneralPrefixSettings do
+
+  let(:migration) { RemoveGeneralPrefixSettings.new }
+
+  def silent
+    verbose = ActiveRecord::Migration.verbose = false
+
+    yield
+
+    ActiveRecord::Migration.verbose = verbose
+  end
+
+  around do |test|
+    silent { test.run }
+  end
+
+  context 'up' do
+    before { migration.down }
+
+    it 'has no settings' do
+      Setting.destroy_all
+      expect(Setting.count).to eq 0
+    end
+
+    it 'removes general key prefix from existing settings' do
+      expect(Setting.exists?(key: 'general_country_source_whitelist')).to eq true
+      expect(Setting.exists?(key: 'general_ip_whitelist')).to eq true
+      expect(Setting.count).to eq 2
+
+      migration.up
+
+      expect(Setting.exists?(key: 'country_source_whitelist')).to eq true
+      expect(Setting.exists?(key: 'ip_whitelist')).to eq true
+
+      migration.down
+    end
+
+    it 'has just new settings' do
+      migration.up
+
+      expect(Setting.exists?(key: 'country_source_whitelist')).to eq true
+      expect(Setting.exists?(key: 'ip_whitelist')).to eq true
+      expect(Setting.count).to eq 2
+
+      migration.down
+    end
+
+    it 'has old and new settings and the old one are getting deleted' do
+      Setting.create(
+        key: 'country_source_whitelist',
+        value: %w[CH DE],
+        type: Setting::CountryCode
+      )
+      Setting.create(
+        key: 'ip_whitelist',
+        value: %w[0.0.0.0 192.168.10.0],
+        type: Setting::IpRange
+      )
+
+      expect(Setting.exists?(key: 'country_source_whitelist')).to eq true
```

Comment:
strange ... all 4 settings exist in that case?

Reply:
settings with new keys are in fixtures.
where do the general prefixed setting come from?